server/routes/room-routes.js

```diff
@@ -828,15 +828,13 @@ router.get(
       }),
     ]);
 
-    // Only `world_readable` or `shared` rooms that are `public` are viewable in the archive
-    const allowedToViewRoom =
-      roomData.historyVisibility === 'world_readable' ||
-      (roomData.historyVisibility === 'shared' && roomData.joinRule === 'public');
```

Comment:
We plan to move ahead with this PR to remove `shared` public rooms from the archive :fast_forward: 

The points in favor of keeping `shared` accessible can be summarized by the following but in the end, while the archive bot was respectful of the technical implications and doesn't expose messages to any further audience (random people), there is a social obligation to consider. This option has been represented as "members only" in many clients which doesn't leave any nuance.

Otherwise, the main idea was if I can view the messages from a Matrix client as a random user, I should also be able to see the messages in the archive. In both the native Matrix client and archive cases, it’s the same result when a random user wants to view a `shared` room:

 - A random Matrix accesses the room, they see the history
 - A random user accesses the room in the archive, they see the history
 - Search engines are not allowed in either case (that only applies to `world_readable` rooms)

The join is mostly a technical detail to anyone trying to view the room. While I don't think the join event provides much value to the room in the normal cases, it could have benefit in tracing bad actors for moderation.

From the [spec](https://spec.matrix.org/v1.7/client-server-api/#room-history-visibility):
> - `world_readable` - All events while this is the `m.room.history_visibility` value may be shared by any participating homeserver with anyone, regardless of whether they have ever joined the room.
> - `shared` - Previous events are always accessible to newly joined members. All events in the room are accessible, even those sent when the member was not a part of the room.

Removing `shared` rooms, does mean we’re re-introducing friction for a portion people which the archive eliminates (which homeserver do I choose, which client, why do I even need an account, how do I view this on mobile, how do I reference and share this message to someone not already in the Matrix ecosystem, etc). But people can update their room to be `world_readable` as they see fit now to regain these benefits for their community.